src/Magnum/Platform/WindowlessEglApplication.h

```diff
@@ -276,16 +276,21 @@ class WindowlessEglContext::Configuration {
          * @m_since_latest
          *
          * When set, the created context will share a subset of OpenGL objects
-         * with @p context, instead of being independent. Many caveats and
+         * with @p context whose associated display is @p display, instead of being independent. 
+         * Many caveats and
          * limitations apply to shared OpenGL contexts, please consult the
          * OpenGL specification for details. Default is `EGL_NO_CONTEXT`, i.e.
          * no sharing.
+         *
+         * @note @p display must be provided to avoid errors on destruction, since 
+         * the display is shared among shared-context on EGL.
          * @see @ref WindowlessEglContext::glContext(),
          *      @ref WindowlessEglApplication::glContext()
          * @requires_gles Context sharing is not available in WebGL.
          */
-        Configuration& setSharedContext(EGLContext context) {
+        Configuration& setSharedContext(EGLContext context, EGLDisplay display) {
```

Comment:
I'd make the display first and context second here, also reorder the getters and private members the same way -- those two are always passed as display first, context second to other EGL functions so this should follow that.

Reply:
Corrected as well